src/tools/lldump.c

```diff
@@ -322,80 +499,244 @@ BOOLEAN tf_print_block(BTREE btree, BLOCK block, void *param)
 	print_block(btree, block, &offset);
 	return TRUE;
 }
+
 /*===============================
  * print_block -- print BLOCK to stdout
  *=============================*/
 void print_block(BTREE btree, BLOCK block, INT32 *offset)
 {
 	INT n;
 	INT32 size=0;
+	INT NRcount = 0;
 
+	printf("\n");
 	/* Step 1: Get length of file */
-	size = (INT32)getfilesize(dbname, fkey2path(block->ix_self));
+	size = (INT32)getfilesize(dbname, fkey2path(ixself(block)));
 
 	/* Step 2: Dump BLOCK structure */
-	printf("BLOCK - DIRECTORY\n");
-	printf(FMT_INT32_HEX ": ix_self: " FMT_INT32_HEX " (%s)\n", *offset, block->ix_self, fkey2path(block->ix_self));
-	*offset += sizeof(block->ix_self);
+	printf("BLOCK - DIRECTORY %s\n",fkey2path(ixself(block)));
+        // BLOCK header
+	printf(FMT_INT32_HEX ":ix_self:   " FMT_INT32_HEX " (%s)\n", *offset, 
+		ixself(block), fkey2path(ixself(block)));
+	*offset += sizeof(ixself(block));
 
-	printf(FMT_INT32_HEX ": ix_type: " FMT_INT32 " (%s)\n", *offset, block->ix_type,
-               (block->ix_type == 1 ? "INDEX" : (block->ix_type == 2 ? "BLOCK" : "UNKNOWN")));
-	*offset += sizeof(block->ix_type);
+	printf(FMT_INT32_HEX ":ix_type:   " FMT_INT32 " (%s)\n", *offset, 
+		ixtype(block), (ixtype(block) == 1 ? "INDEX" : 
+		(ixtype(block) == 2 ? "BLOCK" : "UNKNOWN")));
+	*offset += sizeof(ixtype(block));
 
 #if __WORDSIZE != 16
-	printf(FMT_INT32_HEX ": ix_pad1: " FMT_INT16_HEX "\n", *offset, block->ix_pad1);
+	printf(FMT_INT32_HEX ":ix_pad1:   " FMT_INT16_HEX "\n",
+		*offset, block->ix_pad1);
 	*offset += sizeof(block->ix_pad1);
 #endif
-
-	printf(FMT_INT32_HEX ": ix_parent: " FMT_INT32_HEX " (%s)\n", *offset, block->ix_parent, fkey2path(block->ix_parent));
-	*offset += sizeof(block->ix_parent);
-
-	printf(FMT_INT32_HEX ": ix_nkeys: %d\n", *offset, block->ix_nkeys);
-	*offset += sizeof(block->ix_nkeys);
-
-	for (n=0; n<NORECS; n++) {
-		printf(FMT_INT32_HEX ": ix_rkey[" FMT_INT_04 "]: '%-8.8s'\n", *offset, n, (char *)&block->ix_rkeys[n]);
-		*offset += sizeof(block->ix_rkeys[n]);
+	printf(FMT_INT32_HEX ":ix_parent: " FMT_INT32_HEX " (%s)\n",
+		*offset, ixparent(block), fkey2path(ixparent(block)));
+	*offset += sizeof(ixparent(block));
+
+	NRcount = nkeys(block);
+	printf(FMT_INT32_HEX ":ix_nkeys:  " FMT_INT "\n", *offset, NRcount);
+	*offset += sizeof(nkeys(block));
+
+	// print out block Directory Keys
+	for (n=0; n<NRcount; n++) {
+		check_rkey(&akey,&rkeys(block,n));
+		printf(FMT_INT32_HEX ":ix_rkey[" FMT_INT_04 "]: '%-8.8s'\n",
+			*offset, n, (char *)&rkeys(block,n));
+		*offset += sizeof(rkeys(block,n));
+	}
+	if (NRcount != NORECS) {
+		printf(FMT_INT32_HEX "-" FMT_INT32_HEX 
+			":ix_rkey[" FMT_INT_04 "-" FMT_INT_04 
+			"] default value 0x0000000000000000\n",
+			*offset,
+			*offset + (INT32)((NORECS-NRcount)*sizeof(rkeys(block,0)) - 1),
+			NRcount,(INT)NORECS-1);
+		for (n=NRcount; n<NORECS; n++) {
```

Comment:
I think it would be good to print the actual data, regardless, but we can flag any entry that isn't "default".

Reply:
I was trying to reduce the file sizes by removing unused, zero initialized data, but it only amount to 8% overall
but I still like it.  OTOH I added a -s option, to skup unused zero filled data. So default is to show all data,
and -as will behave like the first commit